implementation/src/main/java/io/smallrye/config/SmallRyeConfig.java

```diff
@@ -191,13 +191,75 @@
+     */
+    @Experimental("Extension to retrieve mandatory sub properties as a Map")
+    public <K, V> Map<K, V> getValues(String name, Class<K> kClass, Class<V> vClass) {
+        final Map<K, V> result = valuesAsMap(name, requireConverter(kClass), requireConverter(vClass));
+        if (result == null) {
+            throw new NoSuchElementException(ConfigMessages.msg.propertyNotFound(name));
+        }
+        return result;
+    }
+
+    /**
+     * Return the content of the direct sub properties as the requested type of Map.
+     *
+     * @param name The configuration property name
+     * @param keyConverter The converter to use for the keys.
+     * @param valueConverter The converter to use for the values.
+     * @param <K> The type of the keys.
+     * @param <V> The type of the values.
+     * @return the resolved property value as an instance of the requested Map or {@code null} if it could not be found.
+     * @throws IllegalArgumentException if a key or a value cannot be converted to the specified types
+     */
+    public <K, V> Map<K, V> valuesAsMap(String name, Converter<K> keyConverter, Converter<V> valueConverter) {
+        final String prefix = name.endsWith(".") ? name : name + ".";
+        final Map<K, V> result = new HashMap<>();
+        for (String propertyName : getPropertyNames()) {
+            if (propertyName.startsWith(prefix)) {
+                final String key = propertyName.substring(prefix.length());
+                if (key.indexOf('.') >= 0) {
+                    // Ignore sub namespaces
+                    continue;
+                }
+                result.put(convertValue(propertyName + "#key", key, keyConverter),
+                        convertValue(propertyName + "#value", getRawValue(propertyName), valueConverter));
+            }
+        }
+        return result.isEmpty() ? null : result;
+    }
+
     /**
      * 
      * This method handles calls from both {@link Config#getValue} and {@link Config#getOptionalValue}.<br>
      */
-    @SuppressWarnings("unchecked")
     public <T> T getValue(String name, Converter<T> converter) {
-        ConfigValue configValue = getConfigValue(name);
+        return getValue(name, getConfigValue(name), converter);
+    }
+
+    /**
+     * @param name The configuration property name
+     * @param configValue The value of the configuration property
+     * @param converter The converter to use to convert the value into the expected type.
+     * @param <T> the expected type.
+     * @return The value of the property converted into the expected type
+     * @throws IllegalArgumentException if the property cannot be converted to the specified type
+     * @throws NoSuchElementException if the property is not defined or is defined as an
+     *         empty string or the converter returns {@code null}
+     */
+    @SuppressWarnings("unchecked")
+    private <T> T getValue(String name, ConfigValue configValue, Converter<T> converter) {
```

Comment:
I think we don't need to split the method anymore, since it only has one usage now.